src/wallet/abstract-wallet.ts

```diff
@@ -795,13 +826,67 @@ abstract class AbstractWallet extends EventEmitter {
     }
   }
 
+  /**
+   * Searches for creation tree height for given merkletree.
+   * @param merkletree - MerkleTree
+   * @param latestTree - number
+   */
+  async getCreationTreeHeight(
+    merkletree: MerkleTree,
+    latestTree: number,
+    creationBlockNumber: number,
+  ): Promise<Optional<number>> {
+    if (creationBlockNumber == undefined) {
+      return undefined;
+    }
+
+    // Loop through each tree and search commitments for creation tree height that matches block number
+    for (let tree = 0; tree <= latestTree; tree += 1) {
+      // Create sparse array of tree
+      // eslint-disable-next-line no-await-in-loop
+      const treeHeight = await merkletree.getTreeLength(tree);
+      const fetcher = new Array<Promise<Optional<Commitment>>>(treeHeight);
+
+      // Fetch each leaf we need to search
+      for (let index = 0; index < treeHeight; index += 1) {
+        fetcher[index] = merkletree.getCommitment(tree, index);
+      }
+
+      // Wait until all leaves are fetched
+      // eslint-disable-next-line no-await-in-loop
+      const leaves = await Promise.all(fetcher);
+
+      // TODO: Binary search all commitments until we find the closest index >= creationBlockNumber
+      // Search through leaves for matching blockNumber
+      if (leaves) {
+        const creationBlockIndex = leaves.findIndex(
+          (commitment) =>
+            commitment &&
+            commitment.blockNumber !== undefined &&
+            commitment.blockNumber >= creationBlockNumber,
+        );
+
+        if (creationBlockIndex > -1) {
+          return creationBlockIndex;
+        }
+      }
+    }
+
+    return undefined;
+  }
+
   /**
    * Clears balances scanned from merkletrees and stored to database.
    * @param chain - chain type/id to clear
    */
   async clearScannedBalances(chain: Chain) {
+    const walletDetails = await this.getWalletDetails(chain);
+    walletDetails.treeScannedHeights = [];
```

Comment:
Nice. Can we actually reconstruct walletDetails to make sure we don't miss a future field? 

Ie. 

```
const walletDetails = await this.getWalletDetails(chain);

const resetWalletDetails: WalletDetails = {
    treeScannedHeights: [], // Clear scanned heights
    creationTreeHeight: walletDetails.creationTreeHeight,
}

```

Reply:
This way, if we add a field in the future, lint will alert us we need to modify this struct.